biodatasets/msh_wsd/msh_wsd.py

```diff
@@ -0,0 +1,245 @@
+            description="MSH-WSD BigBio schema",
+            schema="bigbio_kb",
+            subset_id="msh_wsd",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "msh_wsd_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "ambiguous_word": datasets.Value("string"),
+                    "sentences": [
+                        {
+                            "pmid": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                            "label": datasets.Value("string"),
+                        }
+                    ],
+                    "choices": [
+                        {
+                            "label": datasets.Value("string"),
+                            "concept": datasets.Value("string"),
+                        }
+                    ],
+                }
+            )
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        if self.config.data_dir is None:
+            raise ValueError("This is a local dataset. Please pass the data_dir kwarg to load_dataset.")
+        else:
+            data_dir = data_dir = dl_manager.download_and_extract(self.config.data_dir)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_dir": Path(data_dir),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, data_dir: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        data_dir = data_dir / "MSHCorpus"
+        concepts = data_dir / "benchmark_mesh.txt"
+        with concepts.open() as f:
+            concepts = f.readlines()
+        concepts = [x.strip().split("\t") for x in concepts]
+
+        concept_map = {cuis[0]: {f"M{num}": cui for num, cui in enumerate(cuis[1:], 1)} for cuis in concepts}
+
+        files = list(data_dir.glob("*arff"))
+        for guid, file in enumerate(files):
+            if self.config.schema == "source":
+                for example in self._parse_document(concept_map, file):
+                    yield guid, example
+
+            elif self.config.schema == "bigbio_kb":
+                for document in self._parse_document(concept_map, file):
+                    for example in self._source_to_kb(document):
+                        yield example["id"], example
+
+    def _parse_document(self, concept_map, file: Path):
+
+        with file.open() as f:
```

Comment:
can we open in read or read binary model (depending on what happens with the decoding)?

Reply:
Replacing L188 with `with file.open(mode="r", encoding="iso-8859-1") as f:` should fix this.